### ISSUE TITLE
Get authorization call leads to gateway timeout

### ISSUE THREAD

**Issue by sipe90** [grindsa/acme2certifier#77]
I have an issue with cert-manager and DNS challenges with acme2proxy. When I try to issue a certificate for a service in domain `example.com` which does not run in port 80, the get authorization endpoint times out because acme2certifier tries to solve a HTTP challenge by trying to connect to `http://example.com/.well-known/acme-challenge/xxxxxxxxxxxxxxxxxxx)`. This hangs the request for long which causes a gateway timeout.

It prevents cert-manager from functioning, because when it tries to sync the Challenge-resource, cert-manager makes a getAuthorization request (`https://acme-server.com/acme/authz/xxxxxxxxx`). The call has a 10s timeout and so the sync just fails with "context deadline exceeded".

Is the ACME server supposed to try to solve the challenge when requesting the authorization information and does it have to try to do it with challenge types other than the requested DNS?

**Comment by grindsa** [comment]
Can you please enable debugging in `acme_srv.cfg` and send me the logfile?

Which cert-manager version do you use when observing the issue?

**Comment by sipe90** [comment]
I have sent the logs to via email. I am using the latest version of cert-manager: v1.6.1

**Comment by grindsa** [comment]
I had a deeper look into it. The cert-manager behaviour is a bid weired and differs from the one I see from other acme-clients. After getting the response to the initial authorization request cert-manager continues to poll the auth ressource for whatever reason and this will trigger the validation of all challenges (http-01, dns-01, tls-alpn-01) on a2c which will lead to the observed behaviour. Such behaviour is not wrong; its seems that cert-manager need quite a bid of time for challenge provisioning and to pass some internal self-checks. 

Adapting to this can have some ugly side-effects with other clients. Thus, I choosed a less disruptive approach hoping this will work for you. Changes had been included in the same build as the fix for #76. Please try as well and let me know if situation improves. If not I can disable the challenge validation kicking in when accessing the auth ressource but this needs some careful testing before being released in master.

**Comment by sipe90** [comment]
The devel-version ends up with the same result:
```
I1115 13:09:59.722688       1 controller.go:153] cert-manager/controller/challenges "msg"="syncing item" "key"="xxxxxx"
I1115 13:09:59.722779       1 logger.go:122] cert-manager/acme-middleware "msg"="Calling GetAuthorization"
E1115 13:10:09.725863       1 controller.go:163] cert-manager/controller/challenges "msg"="re-queuing item due to error processing" "error"="context deadline exceeded" "key"="xxxxxx"
```

**Comment by grindsa** [comment]
Ok. Lets go for the more disruptive approach then. As said above I disabled the challenge validation trigger when accessing an auth resource.  Changes are included in a devel build uploaded to docker-hub. Feel free to give it a try.

Did you configure this 10second timeout somewhere? I am asking as I am not able to replicate the issue in my lab when using the same cert-manager version. 
If you don't mind can you share your issuer configuration yaml with me? I am working to extend my regression tests and some configuration files used by others would be extremely helpful as I am not a k8s expert.

**Comment by sipe90** [comment]
I managed to successfully issue the certificate with the latest version. Thank you!

The get authorization call is made here: https://github.com/jetstack/cert-manager/blob/23ee2a2b08fe1522390a10241c25f56254a01ad3/pkg/acme/client/middleware/logger.go#L124
The timeout is defined as 10 seconds in a const at the start of the file.

The issuer configuration is quite simple. I'm running cert-manager in few AWS OpenShift clusters and one EKS cluster. I use Route53 private zones for DNS. The issuer yaml looks something like this:
```
apiVersion: cert-manager.io/v1
kind: ClusterIssuer
metadata:
  name: cluster-issuer
spec:
  acme:
    email: foo@bar.local
    privateKeySecretRef:
      name: acme-proxy
    server: 'https://acme-proxy.com' # Address to acme2certifier
    solvers:
      - http01:
          ingress:
            class: openshift-default # Or "nginx" if using ingress-nginx in Kubernetes
      - dns01:
          route53:
            hostedZoneID: xxxxxxxxxxxxxxxxx
            region: us-west-1 # Value doesn't matter, Route53 is global. Still needs to be set when using ambent credentials (AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY envs set in the cert-manager deployment)
        selector:
          dnsZones:
            - example.com # The domain we want to exclusively use DNS-challenges for, HTTP for everything else
```

**Comment by grindsa** [comment]
Fixes got included in [v0.19](https://github.com/grindsa/acme2certifier/releases/tag/0.19). Thus, I am closing the issue.
